flycheck.el

```diff
@@ -1762,7 +1812,8 @@ nil otherwise."
     (and (flycheck-valid-checker-p checker)
          (not (flycheck-disabled-checker-p checker))
          (flycheck-checker-supports-major-mode-p checker major-mode)
-         (funcall predicate))))
+         (funcall predicate)
+         (flycheck-may-enable-checker checker))))
```

Comment:
I think we should move this before the `predicate` check, and we can merge it with the disable check above.